google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionOptionsTest.java

```diff
@@ -148,6 +149,27 @@ public void testBuildWithAutoConfigEmulator() {
     assertTrue(options.isUsePlainText());
   }
 
+  @Test
+  public void testBuildWithRouteToLeader() {
+    ConnectionOptions.Builder builder = ConnectionOptions.newBuilder();
+    builder.setUri(
+        "cloudspanner:/projects/test-project-123/instances/test-instance-123/databases/test-database-123?routeToLeader=false");
+    ConnectionOptions options = builder.build();
+    assertEquals(options.getHost(), DEFAULT_HOST);
+    assertEquals(options.getProjectId(), "test-project-123");
+    assertEquals(options.getInstanceId(), "test-instance-123");
+    assertEquals(options.getDatabaseName(), "test-database-123");
+    assertFalse(options.isRouteToLeader());
```

Comment:
Can we define `private static final` variables for all such constants instead of defining them inline? We can either have it within this class or a new class in case we see these constants to be generic enough to be reused across different test classes.

This may be ok in case the project ID, instance ID and database name changed with every unit test but I don't see that to be the case here.

Reply:
This is done. And I've used them in this method for now. Refactoring other methods would be unrelated to this PR, and can be taken up later.